### DISCUSSION TITLE
Shaping a mongolian script preceded by non-mongolian characters

### DISCUSSION THREAD

**Discussion by mongol-surah** [harfbuzz/harfbuzz#4156] [General] [answered]
Hi there! 

I've started exploring shaping Mongolian scripts using _harfbuzz_ recently. A seemingly strange thing I encountered is, when a Mongolian word is preceded by non-Monoglian characters, for example, "Aᠲᠤᠷᠤ" and "A ᠲᠤᠷᠤ", the Mongolian word is not shaped correctly - the characters don't connect to each other (but "1ᠲᠤᠷᠤ" and "ᠲᠤᠷᠤA" are shaped perfectly). 

Disconnected shapes generated by `hb-view`:
<img src="https://user-images.githubusercontent.com/118827982/223938433-7bd032ab-ea8d-4654-b716-0ee71e7dbad4.png" width=200></img>

What could be the way to avoid this? I'm a newbie, and I'm sorry if this is a trivial question.

**Comment by behdad** [comment, marked as answer]
HarfBuzz, and hb-view / hb-shape, only shape one direction/script/language at a time. So you cannot mix Latin and Mongolian. Higher-level libraries have to handle such mixing by breaking down the text into separate runs and shape each with HarfBuzz.  Libraries like libraqm or Pango do that.

**Reply by mongol-surah** [reply]
Hi Behdad,
It's good to learn this! Thank you for the reply! 😀
